api-put-object.go

```diff
@@ -170,6 +170,33 @@ func (c Client) putObjectNoChecksum(bucketName, objectName string, reader io.Rea
 		return 0, ErrEntityTooLarge(size, maxSinglePutObjectSize, bucketName, objectName)
 	}
 
+	if size <= -1 {
```

Comment:
We need not do this check and use io.LimitReader() to limit the file, instead we can just wait for the server to return error. Because I think GCS allows much bigger upload in single PUT ( > 5GB ) ... But I am not sure, not able to find documentation. We can check by uploading a large object to GCS and see if it allows > 5GB in single PUT